compiler/rustc_passes/src/check_attr.rs

```diff
@@ -717,6 +717,41 @@ impl CheckAttrVisitor<'tcx> {
         true
     }
 
+    /// Checks that `doc(test(...)` attribute contains only valid attributes. Returns `true` if
+    /// valid.
+    fn check_test_attr(&self, meta: &NestedMetaItem) -> bool {
+        let mut is_valid = true;
+        if let Some(metas) = meta.meta_item_list() {
+            for i_meta in metas {
+                match i_meta.name_or_empty() {
+                    sym::attr | sym::no_crate_inject => {}
+                    _ => {
+                        self.tcx
+                            .sess
+                            .struct_span_err(
```

Comment:
This is wrong. It needs to be a future-incompat lint, not a hard error. This is the exact issue from https://github.com/rust-lang/rust/pull/82702, please be more careful in the future.

Reply:
Some day I'll remember about the future-incompat lint... Good catch!